File: ulohy/16-kosodelnik/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let side, height;

rl.on("line", (line) => {
    if (side === undefined) {
        side = parseInt(line);
        return;
    }
    height = parseInt(line);
    
    printRhomboid(side, height);

    rl.close();
});

function printRhomboid(side, height) {
    for (let i = 0; i < height; i++) {
        let line = "";
        for (let j = 0; j < height - i - 1; j++) {
            line += " ";
        }
        for (let j = 0; j < side; j++) {
            const output = (i > 0 && i < height - 1 && j > 0 && j < side - 1) ? " " : "#";
            line += output;
        }
        console.log(line);
    }
}

File: ulohy/16-kosodelnik/reseni/main.py
def print_rhomboid(side, height):
    for i in range(height):
        for j in range(height - i - 1):
            print(" ", end="")

        for j in range(side):
            output = " " if (i > 0 and i < height - 1 and j > 0 and j < side - 1) else "#"
            print(output, end="")

        print()


if __name__ == "__main__":
    side = int(input())
    height = int(input())

    print_rhomboid(side, height)
